app/routes/index.tsx

```diff
@@ -8,97 +8,84 @@ import Calendar from '~/components/Calendar';
-        data: null,
-        ENV: baseUrls,
-        error: 'Unable to fetch the owner details, please login',
-      }));
-    const ownerId = calendarData?.data?.rCal[0]?.ownerId;
-    const calendarId = calendarData?.data?.rCal[0]?.id;
+  // no session cookie
+  return redirect('/login');
+};
 
-    if (ownerId) {
-      const { data: eventDetails } = await axios
-        .get(getEvents(process.env.API_HOST!, calendarId, startTime, endTime))
-        .catch((_) => ({
-          data: null,
-          ENV: baseUrls,
-          error: 'Unable to fetch the event Details',
-        }));
+function CalendarPage() {
+  const { setEvents, events, view } = useStore((state) => state);
+  const { apiHost } = useLoaderData();
 
-      return json<LoaderData>({ events: eventDetails?.data, calendarId, ENV: baseUrls, error: null });
-    }
+  const fetchEvents = async () => {
+    if (!apiHost) return;
 
-    return json<any>({ events: null, ENV: baseUrls, error: 'Unable to fetch the owner details' });
-  } catch (error) {
-    return { events: null, ENV: baseUrls, error };
-  }
-};
+    let calendarId = 0;
+    const startTime = dayjs().subtract(1, 'months').startOf('month').unix() * 1000;
+    const endTime = dayjs().add(1, 'months').endOf('month').unix() * 1000;
 
-function CalendarPage() {
-  const { setEvents, events: eventList, view } = useStore((state) => state);
-  const { events, error, calendarId } = useLoaderData();
-  useEffect(() => {
-    if (error === null && events.length > 0) {
-      // TODO: show a  different message if events are not present in the given date range
-      localStorage.setItem('calendarId',calendarId)
-      setEvents([parseEvents(events)][0]);
-    } else if (error === null && events.length === 0) {
-      // TODO: discuss regarding display of user message in case of no events
-      localStorage.setItem('calendarId',calendarId)
-    } else {
-      // TODO: redirect the user to login page on 401
-      toast.error(error, {
+    if (typeof window !== 'undefined') {
+      const cookie = parseCookie(document.cookie);
+      calendarId = Number(cookie.calendarId);
+    }
+
+    try {
+      const { data: eventsList } = await axios.get(
+        getEvents(apiHost, calendarId, startTime, endTime),
+        {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          withCredentials: true,
+        },
+      );
+      setEvents(parseEvents(eventsList.data));
+    } catch (error) {
+      toast.error('Unable to fetch the event Details', {
         toastId: 'events_error',
       });
     }
+  };
+
+  useEffect(() => {
+    fetchEvents();
   }, []);
 
   return (
     <div className="">
       <div className="flex flex-col-reverse md:flex-row ">
         <Navbar />
         <div className="flex justify-center flex-grow">
-          <Calendar events={eventList.length ? eventList : parseEvents(events ?? [])} view={view} />
+          <Calendar events={events} view={view} />
```

Comment:
Are we handling scenarios when eventList is undefined?

Reply:
@harshith-venkatesh by default its an empty array and it wont be undefined as we only set it when we get it, else we toast the error.

Reply:
Ok great